### PR TITLE
Concurrent table deletion and schema modification should be reported as a schema conflict.

### PR DESCRIPTION
**Pull request by nicktobey**
Previously during a merge, if one branch altered a table schema and the other branch deleted the table, this would be an error that would automatically abort the merge. Now it's a schema conflict.

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998097 to 99.998097</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>37ae342</TD>
<TD>not ok</TD>
<TD>113</TD>
</TR>
<TR><TD>37ae342</TD>
<TD>ok</TD>
<TD>5937344</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>37ae342</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998097</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998097 to 99.998097</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ce9894a</TD>
<TD>not ok</TD>
<TD>113</TD>
</TR>
<TR><TD>ce9894a</TD>
<TD>ok</TD>
<TD>5937344</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ce9894a</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998097</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998097</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>7c11c6a</TD>
<TD>not ok</TD>
<TD>113</TD>
</TR>
<TR><TD>7c11c6a</TD>
<TD>ok</TD>
<TD>5937344</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>7c11c6a</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998097</TD>
</TR>
</table>

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998097</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>1fedd61</TD>
<TD>not ok</TD>
<TD>113</TD>
</TR>
<TR><TD>1fedd61</TD>
<TD>ok</TD>
<TD>5937344</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>1fedd61</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998097</TD>
</TR>
</table>

File: go/libraries/doltcore/sqle/dtables/schema_conflicts_table.go [hunk 1]
```diff
@@ -162,19 +163,47 @@ func newSchemaConflict(ctx context.Context, table string, baseRoot *doltdb.RootV
...
-		return schemaConflict{}, err
+	var base string
+	if baseSch != nil {
+		var err error
+		base, err = getCreateTableStatement(table, baseSch, baseFKs, bs)
+		if err != nil {
+			return schemaConflict{}, err
+		}
+	} else {
+		base = "<deleted>"
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dtables/schema_conflicts_table.go:174]: Need tests for new table contents in the three cases shown here

**Review by zachmu — APPROVED** [review]
LG, only comment is apparently missing test for new behavior of conflicts table

File: go/libraries/doltcore/merge/merge_schema.go [hunk 2]
```diff
@@ -91,6 +96,9 @@ func (sc SchemaConflict) messages() (mm []string) {
 	for _, c := range sc.ChkConflicts {
 		mm = append(mm, c.String())
 	}
+	if sc.ModifyDeleteConflict {
+		mm = append(mm, "table was modified in one branch and deleted in the other")
```
[the remark this answers is not in the document]
Comment (nicktobey) [reply go/libraries/doltcore/merge/merge_schema.go:100]: This comment made me realize that I was treating data change and schema change the same here when we should be distinguishing them and treating them differently. I've fixed that and also added a bunch more tests.

File: go/libraries/doltcore/sqle/dtables/schema_conflicts_table.go (hunk 1 above)
Comment (nicktobey) [reply go/libraries/doltcore/sqle/dtables/schema_conflicts_table.go:174]: Added.

**Comment by coffeegoddd** [comment]
@nicktobey __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998653 to 99.998653</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>7e5f0ea</TD>
<TD>not ok</TD>
<TD>80</TD>
</TR>
<TR><TD>7e5f0ea</TD>
<TD>ok</TD>
<TD>5937377</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>7e5f0ea</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998653</TD>
</TR>
</table>

**Pull request merged**